source/cppfront.cpp

```diff
@@ -1874,21 +1873,25 @@ class cppfront
                 }
 
                 if (i->id_expr) {
-                    auto print = std::string{};
-                    printer.emit_to_string(&print);
-                    emit(*i->id_expr, false /*not a local name*/);
-                    printer.emit_to_string();
+
+                    if (args) {
+                        // if args are stored it means that this is function or method
+                        // that is not handled by UFCS e.g. that has more than one template argument
+                        suffix.emplace_back(")", n.position());
+                        for (auto&& e: args.value()) {
+                            suffix.push_back(e);
+                        }
+                        suffix.emplace_back("(", n.position());
+                        args.reset();
+                    }
+
+                    auto print = print_to_string(*i->id_expr, false /*not a local name*/);
```

Comment:
These two `suffix.` lines should also use the positions:

```
                        suffix.emplace_back(")", args_close_pos);
                        ...
                        suffix.emplace_back("(", args_open_pos);
```

Reply:
Ok. Thank you for all your comments. I will introduce these changes.